lib/screen/feralfile_home/artwork_view.dart

```diff
@@ -0,0 +1,375 @@
+import 'package:feralfile_app_theme/feral_file_app_theme.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+class ExploreSeriesView extends StatefulWidget {
+  final String? searchText;
+  final Map<FilterType, FilterValue> filters;
+  final SortBy sortBy;
+  final int pageSize;
+
+  const ExploreSeriesView(
+      {required this.sortBy,
+      this.searchText,
+      this.pageSize = 20,
+      super.key,
+      this.filters = const {}});
+
+  @override
+  State<ExploreSeriesView> createState() => ExploreSeriesViewState();
+
+  bool isEqual(Object other) =>
+      other is ExploreSeriesView &&
+      other.searchText == searchText &&
+      other.filters == filters &&
+      other.sortBy == sortBy &&
+      other.pageSize == pageSize;
+}
+
+class ExploreSeriesViewState extends State<ExploreSeriesView> {
+  List<FFSeries>? _series;
+  late Paging _paging;
+  late ScrollController _scrollController;
+  bool _isLoading = false;
+
+  void scrollToTop() {
+    unawaited(_scrollController.animateTo(
+      0,
+      duration: const Duration(milliseconds: 300),
+      curve: Curves.easeInOut,
+    ));
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController = ScrollController();
+    _scrollController.addListener(() {
+      if (_scrollController.position.pixels + 100 >
+          _scrollController.position.maxScrollExtent) {
+        unawaited(_loadMoreSeries(context,
+            offset: _paging.offset + _paging.limit, pageSize: _paging.limit));
+      }
+    });
+    _paging = Paging(offset: 0, limit: widget.pageSize, total: 0);
+    unawaited(
+        _fetchSeries(context, offset: _paging.offset, pageSize: _paging.limit));
+  }
+
+  @override
+  void didUpdateWidget(covariant ExploreSeriesView oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (!oldWidget.isEqual(widget) || true) {
+      unawaited(_fetchSeries(context));
+    }
+  }
+
+  @override
+  void dispose() {
+    _scrollController.dispose();
+    super.dispose();
+  }
+
+  Widget _loadingView(BuildContext context) => const Padding(
+        padding: EdgeInsets.only(bottom: 100),
+        child: LoadingWidget(),
+      );
+
+  Widget _emptyView(BuildContext context) {
+    final theme = Theme.of(context);
+    return Center(
+      child:
+          Text('no_series_found'.tr(), style: theme.textTheme.ppMori400White14),
+    );
+  }
+
+  Widget _seriesView(BuildContext context, List<FFSeries> series) => SeriesView(
+        series: series,
+        scrollController: _scrollController,
+        padding: const EdgeInsets.only(bottom: 100),
+      );
+
+  @override
+  Widget build(BuildContext context) {
+    if (_series == null) {
+      return _loadingView(context);
+    } else if (_series!.isEmpty) {
+      return _emptyView(context);
+    } else {
+      return Expanded(child: _seriesView(context, _series!));
+    }
+  }
+
+  Future<List<FFSeries>> _fetchSeries(BuildContext context,
```

Comment:
it is unusual that you need to have both `_fetchSeries` and `_loadMoreSeries`.
can you just making one to appendSeries. Beside, you can use infinite paging, it's already been used in our app